components/core/src/streaming_compression/passthrough/Decompressor.cpp

```diff
@@ -5,51 +5,90 @@
 
 namespace streaming_compression { namespace passthrough {
     ErrorCode Decompressor::try_read (char* buf, size_t num_bytes_to_read, size_t& num_bytes_read) {
-        if (nullptr == m_compressed_data_buf) {
-            return ErrorCode_NotInit;
+        if (InputType::NotInitialized == m_input_type) {
+            throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
         }
         if (nullptr == buf) {
-            return ErrorCode_BadParam;
+            throw OperationFailed(ErrorCode_BadParam, __FILENAME__, __LINE__);
         }
 
-        if (m_compressed_data_buf_len == m_decompressed_stream_pos) {
-            return ErrorCode_EndOfFile;
+        if (InputType::CompressedDataBuf == m_input_type) {
+            if (m_compressed_data_buf_len == m_decompressed_stream_pos) {
+                return ErrorCode_EndOfFile;
+            }
+
+            num_bytes_read = std::min(num_bytes_to_read, m_compressed_data_buf_len - m_decompressed_stream_pos);
+            memcpy(buf, &m_compressed_data_buf[m_decompressed_stream_pos], num_bytes_read);
+        } else if (InputType::File == m_input_type) {
```

Comment:
I personally would prefer to have an "else throw" here just for sanity check

Reply:
Actually i just realized that the only other possiblity will be checked above.. maybe add a comment?

Reply:
Good point. Addressed.